tests/zfs-tests/tests/perf/perf.shlib

```diff
@@ -122,33 +122,33 @@ function do_fio_run_impl
 	# disable client cache for reads.
 	if [[ $NFS -eq 1 ]]; then
 		export DIRECT=1
-		do_setup_nfs $script
+		do_setup_nfs "$script"
 	else
 		export DIRECT=0
 	fi
 
 	# This will be part of the output filename.
-	typeset suffix=$(get_suffix $threads $sync $iosize)
+	typeset suffix=$(get_suffix "$threads" "$sync" "$iosize")
 
 	# Start the data collection
-	do_collect_scripts $suffix
+	do_collect_scripts "$suffix"
 
 	# Define output file
 	typeset logbase="$(get_perf_output_dir)/$(basename \
-	    $SUDO_COMMAND)"
+	    "$SUDO_COMMAND")"
```

Comment:
It seems you are adding quotes inside quotes. Doesn't it need escaping, or something?

Reply:
This is using a subshell command, the contents of $() are evaluated and substituted in to the statement before its evaluated as a whole. So escaping here would actually result in incorrect behavior.